autograder/submitty_autograding_shipper.py

```diff
@@ -948,6 +960,40 @@ def checkout_vcs_repo(config, my_file):
                 #    # and check out the right version
                 #    subprocess.call(['git', 'checkout', '-b', 'grade', what_version])
 
+                # copy the subdirectory we want to the
+                # original checkout path and remove the extra files
+                if subdirectory_grading:
+                    try:
+                        vcs_subdirectory = vcs_subdirectory[1:]
+                        files = os.listdir(os.path.join(sub_checkout_path, vcs_subdirectory))
+                        if files:
```

Comment:
is there a reason we need to do this one file at a time?  can we not just move the subtree in one command?  does this work for a subdirectory with a tree of subdirectories or only a flat list of files in a subdirectory?  this would be unnecessarily expensive for a subdirectory with LOTs of files.

note: I haven't tested yet and I don't remember the details of the python file functions.